Makefile

```diff
@@ -73,28 +66,72 @@ test-verbose: ARGS=-v
 test-race:    ARGS=-race
 $(TEST_TARGETS): test
 test tests:
-	$Q $(GO) test -timeout $(TIMEOUT)s $(ARGS) $(TESTPKGS)
+	${call for-all-modules,$(GO) test -timeout $(TIMEOUT)s $(ARGS) $(PKGS)}
 
 COVERAGE_MODE    = atomic
 COVERAGE_PROFILE = $(COVERAGE_DIR)/profile.out
 .PHONY: test-coverage
 test-coverage: COVERAGE_DIR := $(TEST_RESULTS)/coverage_$(shell date -u +"%s")
 test-coverage:
 	$Q mkdir -p $(COVERAGE_DIR)
-	$Q $(GO) test \
-		-coverpkg=$$($(GO) list -f '{{ join .Deps "\n" }}' $(TESTPKGS) | \
-					grep '^$(MODULE)/' | \
-					tr '\n' ',' | sed 's/,$$//') \
-		-covermode=$(COVERAGE_MODE) \
-		-coverprofile="$(COVERAGE_PROFILE)" $(TESTPKGS)
+	${call for-all-modules,$(GO) test -coverpkg=$(PKGS) -covermode=$(COVERAGE_MODE) -coverprofile="$(COVERAGE_PROFILE)" $(PKGS)}
 
 .PHONY: lint
 lint: | $(GOLANGCI_LINT)
 # Run once to fix and run again to verify resolution.
-	$Q $(GOLANGCI_LINT) run --fix && $(GOLANGCI_LINT) run
+	${call for-all-modules,$(GOLANGCI_LINT) run --fix && $(GOLANGCI_LINT) run}
+
+# Pre-release targets
+
+.PHONY: add-tag
+add-tag: # example usage: make add-tag tag=v1.100.1
+	$Q [ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	@echo "Adding tag $(tag)"
+	$Q git tag -a $(tag) -s -m "Version $(tag)"
+	$Q set -e; for dir in $(SUBMODULES); do \
+	  (echo Adding tag "$${dir:2}/$(tag)" && \
+	 	git tag -a "$${dir:2}/$(tag)" -s -m "Version ${dir:2}/$(tag)" ); \
+	done
+
+.PHONY: delete-tag
+delete-tag: # example usage: make delete-tag tag=v1.100.1
+	$Q [ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	@echo "Deleting tag $(tag)"
+	$Q git tag -d $(tag)
+	$Q set -e; for dir in $(SUBMODULES); do \
+	  (echo Deleting tag "$${dir:2}/$(tag)" && \
+	 	git tag -d "$${dir:2}/$(tag)" ); \
+	done
+
+.PHONY: push-tag
+push-tag: # example usage: make push-tag remote=origin tag=v1.100.1 commit=<hash>
+	$Q [ "$(remote)" ] || ( echo ">> 'remote' is not set"; exit 1 )
+	$Q [ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	$Q [ "$(commit)" ] || ( echo ">> 'commit' is not set"; exit 1 )
```

Comment:
This looks like it was added in the wrong target. I was thinking specifying the commit explicitly in `add-tag`. I'm pretty sure having it here will not only push the tag but try to push that commit in addition.

Reply:
You are correct. I am a dummy :D

Reply:
Addressed c64ba7d3